src/main/java/io/cryostat/core/net/JFRConnection.java

```diff
@@ -148,6 +161,43 @@ public synchronized int getPort() {
         }
     }
 
+    public synchronized String getJvmId() throws AttributeNotFoundException, InstanceNotFoundException, MBeanException, ReflectionException, IOException {
+        if (!isConnected()) {
+            connect();
+        }
+        List<String> attrNames =
+                new ArrayList<>(
+                        Arrays.asList(
+                                "ClassPath",
+                                "Name",
+                                "InputArguments",
+                                "LibraryPath",
+                                "VmVendor",
+                                "VmVersion",
+                                "StartTime"));
+
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream(512);
+                DataOutputStream dos = new DataOutputStream(baos)) {
+            for (String attr : attrNames) {
+                Object attrObject =
+                        this.rjmxConnection.getAttributeValue(
+                                new MRI(Type.ATTRIBUTE, ConnectionToolkit.RUNTIME_BEAN_NAME, attr));
+
+                if (attrObject instanceof String || attrObject instanceof Long) {
+                    dos.writeUTF(attrObject.toString());
+                } else if (attrObject instanceof String[]) {
```

Comment:
Maybe `attrObject.getClass().isArray()` here so that arrays of things other than `String` get handled the same? It's not critical since it's highly unlikely that the types of any of these bean attributes changes, but still.